### PR TITLE
Make a separate interval loop to send data

### PR DESCRIPTION
**Pull request by 0hoo**

### CODE REVIEW

File: web/app.ts [hunk 1]
```diff
@@ -66,6 +67,25 @@ function startRecording(microphone: MediaStream) {
 
 async function onRecordDataAvailable(blob: Blob) {
   const base64Data = (await blobToBase64(blob)) as string;
+  dataToSend.push(base64Data);
+}
+
+console.log("init");
```
Comment (jinhyung) [inline web/app.ts:73]: Let's remove or:
```suggestion
console.log("Initialize XL8 Skroll E2E client...");
```
?

File: web/app.ts [hunk 2]
```diff
@@ -66,6 +67,25 @@ function startRecording(microphone: MediaStream) {
...
+initE2E("demo", "demo").then((id) => {
+  console.log(`got session id ${id}`);
+  SESSION_ID = id;
+  captureMicrophone(function (mic) {
+    startRecording(mic);
+  });
+});
+
+async function sendData() {
+  const base64Data = dataToSend.shift();
```
Comment (jinhyung) [inline web/app.ts:84]: Shall we add comments about why we are doing this?
```suggestion
  //Speech data has to be sent in order.
  const base64Data = dataToSend.shift();
```

**Review by jinhyung — APPROVED** [review]
LGTM! Thanks for the fix! :)

**Review by dasony — APPROVED** [review]
LGTM! Thank you!
It would be even more efficient if we could merge the blobs and send it in a single request. Not sure if you can append blobs, since we can't just concatenate two base64 strings.. Also not sure how often multiple items exist in dataToSend to warrant doing this.

**Comment by 0hoo** [comment]
> LGTM! Thank you!
> It would be even more efficient if we could merge the blobs and send it in a single request. Not sure if you can append blobs, since we can't just concatenate two base64 strings.. Also not sure how often multiple items exist in dataToSend to warrant doing this.

yeah, last Friday I tried to concatenate blobs, then https://developer.mozilla.org/en-US/docs/Web/API/Blob/slice it, but like I said, I got an error from server with that request. Maybe I did some mistake. I will add it to our backlog.

**Pull request merged**